### PR TITLE
PLT-2496 Adding edit of incoming & outgoing webhooks

### PR DESCRIPTION
**Pull request by mpoornima**
#### Summary
* Added edit functionality for incoming & outgoing webhooks
* For breaking changes when modifying outgoing webhooks, a confirmation dialog appears asking user for affirmation
#### Notes to the reviewer
* Renamed all instances of `errror` to `error`. Hope its okay.  
* The current webapp tests for hooks under `client_hooks.test.jsx` is only checking for the case where hooks are not enabled. I've followed similar approach as I'm not sure how to enable webhooks in the test.
* Refactored couple of common functionalities in `webhook.go`. Hope its okay. 
* I'm not sure about all the permission checks that must be done for updation. I've added as many as I could find from `create` & `delete` functions. Do let me know if I've missed any & if some part of the logic is wrong. 
* Not sure if I have to update the documentation. Do let me know. 

#### Ticket Link
Jira: [PLT-2496](https://mattermost.atlassian.net/browse/PLT-2496)
GitHub: #4753 

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [ ] All new/modified APIs include changes to the drivers
- [ ] Has enterprise changes (please link)
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

Thanks !

### CODE REVIEW

**Comment by jasonblais** [comment]
Hey @mpoornima, looks like there's a build failure. Can you help take a look? If you have any questions about it, don't hesitate to let us know.

Huge thanks for your contribution!!

**Comment by mpoornima** [comment]
@jasonblais fixed the failure. Waiting for the next build.

**Comment by mpoornima** [comment]
@jasonblais Build successful now.

**Comment by jasonblais** [comment]
Awesome, thanks! @doosp and I will test this out

**Comment by doosp** [comment]
@mpoornima 

Really appreciate your interest in our product and contribution

I tested and couldn't find any issue.

@jasonblais  @esethna  would you review when you have a chance? Thanks.

Test Summary:
1.  Change outgoing and incoming webhook. The incoming webhook didn't show any confirmation popup. The outgoing webhook showed the confirmation popup when trigger word, content type, and url.
2. Cancel doesn't  save the update.
3. Update works. If wrong change is made ( e.g. wrong outgoing webhook url ) breaks the webhook. Trigger word change works as well.
4. Token change works correctly.
5. Delete works correctly.

File: api/webhook.go [hunk 1]
```diff
@@ -21,10 +21,12 @@ func InitWebhook() {
 	l4g.Debug(utils.T("api.webhook.init.debug"))
 
 	BaseRoutes.Hooks.Handle("/incoming/create", ApiUserRequired(createIncomingHook)).Methods("POST")
+	BaseRoutes.Hooks.Handle("/incoming/update", ApiUserRequired(updateIncomingHook)).Methods("PUT")
```
Comment (enahum) [inline api/webhook.go]: My guess is that this routes should be `POST` and `PUT` should be use in the API v4, perhaps @jwilander can confirm

File: api/webhook.go [hunk 2]
```diff
@@ -21,10 +21,12 @@ func InitWebhook() {
...
 	BaseRoutes.Hooks.Handle("/incoming/delete", ApiUserRequired(deleteIncomingHook)).Methods("POST")
 	BaseRoutes.Hooks.Handle("/incoming/list", ApiUserRequired(getIncomingHooks)).Methods("GET")
 
 	BaseRoutes.Hooks.Handle("/outgoing/create", ApiUserRequired(createOutgoingHook)).Methods("POST")
+	BaseRoutes.Hooks.Handle("/outgoing/update", ApiUserRequired(updateOutgoingHook)).Methods("PUT")
```
Comment (enahum) [inline api/webhook.go]: same here

File: api/webhook.go (hunk 1 above)
Comment (jwilander) [inline api/webhook.go]: Yes, please make this a POST so that we're consistent with the rest of the APIv3 endpoints

Comment (jasonblais) [inline api/webhook.go]: @mpoornima can you help review the feedback above?

Comment (mpoornima) [inline api/webhook.go]: Hey, I'm on vacation right now and will do it when I return (Feb 27th)

Comment (jasonblais) [inline api/webhook.go]: Thanks @mpoornima for the update!  And no rush on this, just wanted to make sure you weren't blocked on anything.

Hope you have a good vacation!

**Comment by mpoornima** [comment]
@jwilander @enahum Made the requisite changes. Please have a look.

**Comment by jasonblais** [comment]
This is good for 3.7, updated milestone. Thanks @mpoornima!

**Pull request merged**